jurigged/live.py

```diff
@@ -90,6 +97,8 @@ def __init__(self, registry, debounce=DEFAULT_DEBOUNCE, poll=False):
         self.registry.precache_activity.register(self.on_prepare)
         self.debounce = debounce
         self.poll = poll
+        self.prerun_callback = prerun_callback
+        self.postrun_callback = postrun_callback
```

Comment:
This only allows a single callback for prerun and one for postrun, instead you can use `jurigged.utils.EventSource`, which is used like this:

```python
from .utils import EventSource
evt = EventSource(callback1, callback2, ...)
evt.register(callback3)
evt.emit(arg1, arg2, ...)
```

Then we can register multiple callbacks for each of the two events.

Reply:
Sorry for the super late reply on this. I'm a bit confused by this recommendation, as I'm not sure the event would capture the same infomation? I added the other suggested changes but am kind of lost what you mean here

Reply:
Yeah, sorry, that was pretty unclear. Basically, instead of an interface where we give a single callback function to the constructor:

```python
w = Watcher(..., prerun_callback=fn1)
```

We can instead make `prerun_callback` an `EventSource` (let's call it `prerun`) and then:

```python
w = Watcher(...)
w.prerun.register(fn1)
# w.prerun.register(fn2)  # we can register a second callback etc.
```

And the watcher can call `w.prerun.emit(path, cf)` to trigger all callbacks. Jurigged already contains several such events, so it'd be more consistent.

Reply:
Added and tested this, should be good to go now